### PR TITLE
stunnel: Update to 5.35 and use pthreads

### PR DESCRIPTION
**Pull request by diizzyy**
Maintainer: @mhaas

Compile tested: ramips, D-Link DIR-860L B1, LEDE r1242
Run tested: ramips, D-Link DIR-860L B1, LEDE r1242

Description:
Updates stunnel to 5.35 and changes it from using forking to pthreads.

Signed-off-by: Daniel Engberg <daniel.engberg.lists@pyret.net>

### CODE REVIEW

**Comment by karlp** [comment]
https://github.com/openwrt/packages/pull/2544/files does a better job of setting mirrors.  Why did you leave that behind when you update this?

Further, "changing to use pthreads" is obvious from the patch.  What's not obvious is _why_ you changed it from fork to pthreads.   A single commit bumping the version and fixing the mirrors would be much easier to accept.

File: net/stunnel/Makefile [hunk 1]
```diff
@@ -8,16 +8,16 @@
...
-PKG_VERSION:=5.20
+PKG_VERSION:=5.35
 PKG_RELEASE:=1
 
 PKG_LICENSE:=GPL-2.0+
 PKG_MAINTAINER:=Michael Haas <haas@computerlinguist.org>
 PKG_LICENSE_FILES:=COPYING COPYRIGHT.GPL
 
-PKG_SOURCE_URL:=http://stunnel.cybermirror.org/archive/5.x/
+PKG_SOURCE_URL:=http://ftp.nluug.nl/pub/networking/stunnel/
```
Comment (karlp) [inline net/stunnel/Makefile]: what's wrong with just straight up stunnel.org/downloads?

**Comment by diizzyy** [comment]
pthreads are more efficient than forking and you have native support in musl.
As for using stunnel.org, official sites tends to be very slow as people in general are lazy and don't lookup mirrors to offload main sites. stunnel.cybermirror.org has proven to be dodgy for me at least hence the switch.

**Comment by karlp** [comment]
That comment there about pthreads more efficient? That's a _great_ thing to put in the commit comment.  Re mirrors, you know you can use multiple sites? Like in https://github.com/openwrt/packages/pull/2544/files   ?

**Comment by diizzyy** [comment]
I figured it was common knowledge by now and I didn't realize that you wanted that info as there are other commits with even more sparse information than mine that has been accepted.
https://github.com/openwrt/packages/commit/bc91db0d2ace45da485d55dc89ee65d559338ec5

**Comment by karlp** [comment]
Just because some commits are bad doesn't mean you should be as bad.

**Comment by diizzyy** [comment]
@hnyman

**Comment by diizzyy** [comment]
Doesn't seem to be of interest, closing.

**Pull request closed without merging**